hazelcast-client/src/test/java/com/hazelcast/client/cache/nearcache/ClientNearCacheTestSupport.java

```diff
@@ -529,6 +538,112 @@ public void run() throws Exception {
+                .setEvictionPolicy(EvictionPolicy.LRU)
+                .setMaximumSizePolicy(EvictionConfig.MaxSizePolicy.ENTRY_COUNT)
+                .setSize(size);
+
+        NearCacheConfig nearCacheConfig = createNearCacheConfig(inMemoryFormat);
+        nearCacheConfig.setEvictionConfig(evictionConfig);
+
+        NearCacheTestContext context = createNearCacheTest(DEFAULT_CACHE_NAME, nearCacheConfig);
+
+        // populate map with an extra entry
+        for (int i = 0; i < size + 1; i++) {
+            context.cache.put(i, "value-" + i);
+        }
+
+        // populate Near Caches
+        for (int i = 0; i < size; i++) {
+            context.cache.get(i);
+        }
+
+        NearCacheStats statsBeforeEviction = getNearCacheStats(context.cache);
+
+        // trigger eviction via fetching the extra entry
+        context.cache.get(size);
+
+        waitForNearCacheEvictions(context.cache, expectedEvictions);
+
+        // we expect (size + the extra entry - the expectedEvictions) entries in the Near Cache
+        int expectedOwnedEntryCount = size + 1 - expectedEvictions;
+
+        NearCacheStats stats = getNearCacheStats(context.cache);
+        assertEquals("got the wrong ownedEntryCount", expectedOwnedEntryCount, stats.getOwnedEntryCount());
+        assertEquals("got the wrong eviction count", expectedEvictions, stats.getEvictions());
+        assertEquals("got the wrong expiration count", 0, stats.getExpirations());
+        assertEquals("we expect the same hits", statsBeforeEviction.getHits(), stats.getHits());
+        assertEquals("we expect the same misses", statsBeforeEviction.getMisses(), stats.getMisses());
+    }
+
+    protected void testNearCacheExpiration_withTTL(InMemoryFormat inMemoryFormat) {
+        NearCacheConfig nearCacheConfig = createNearCacheConfig(inMemoryFormat);
+        nearCacheConfig.setTimeToLiveSeconds(MAX_TTL_SECONDS);
+
+        testNearCacheExpiration(nearCacheConfig, MAX_TTL_SECONDS);
+    }
+
+    protected void testNearCacheExpiration_withMaxIdle(InMemoryFormat inMemoryFormat) {
+        NearCacheConfig nearCacheConfig = createNearCacheConfig(inMemoryFormat);
+        nearCacheConfig.setTimeToLiveSeconds(MAX_IDLE_SECONDS);
+
+        testNearCacheExpiration(nearCacheConfig, MAX_IDLE_SECONDS);
+    }
+
+    private void testNearCacheExpiration(NearCacheConfig nearCacheConfig, int expireSeconds) {
+        final int size = 147;
+
+        final NearCacheTestContext context = createNearCacheTest(DEFAULT_CACHE_NAME, nearCacheConfig);
+
+        for (int i = 0; i < size; i++) {
+            context.cache.put(i, "value-" + i);
+            context.cache.get(i);
+        }
+
+        final NearCacheStats statsBeforeExpiration = getNearCacheStats(context.cache);
+        assertEquals(format("we expected to have all map entries in the Near Cache (%s)", statsBeforeExpiration),
+                size, statsBeforeExpiration.getOwnedEntryCount() + statsBeforeExpiration.getExpirations());
```

Comment:
I added the expiration count here, since there was a test failure that some entries were already removed from the cache.